lib/ui/view/general_search/general_search_view.dart

```diff
@@ -0,0 +1,263 @@
+                                    size: 20.0,
+                                  )),
+                              title: Text(recentSearch,
+                                  style: AppTextStyle.darkGreySize16),
+                              onTap: () {
+                                searchController.text = recentSearch;
+                                FocusScope.of(context).unfocus();
+                                model.notifyListeners();
+                                model.search(needle);
+                              },
+                              dense: true,
+                            );
+                          },
+                        ),
+                      ],
+                      const SizedBox(height: 8.0),
+                      Padding(
+                        padding: const EdgeInsets.only(left: 12.0),
+                        child: Text(locale.narrowYourSearch,
+                            style: AppTextStyle.darkGreySize14Bold),
+                      ),
+                      const SizedBox(height: 8.0),
+                      ListTile(
+                        horizontalTitleGap: 0.0,
+                        leading: const Icon(
+                          Icons.add_box_outlined,
+                          color: AppColors.zuriDarkGrey,
+                          size: 20.0,
+                        ),
+                        contentPadding:
+                            const EdgeInsets.symmetric(horizontal: 12.0),
+                        trailing: Text(locale.inExample,
+                            style: AppTextStyle.darkGreySize16),
+                        title: Text(locale.inFilter,
+                            style: AppTextStyle.darkGreySize16),
+                        onTap: () {
+                          searchController.text = "${locale.inFilter}#";
+                          searchFocusNode.requestFocus();
+                          model.notifyListeners();
+                        },
+                        dense: true,
+                      ),
+                      ListTile(
+                        horizontalTitleGap: 0.0,
+                        leading: const Icon(
+                          Icons.add_box_outlined,
+                          color: AppColors.zuriDarkGrey,
+                          size: 20.0,
+                        ),
+                        contentPadding:
+                            const EdgeInsets.symmetric(horizontal: 12.0),
+                        trailing: Text(locale.fromExample,
+                            style: AppTextStyle.darkGreySize16),
+                        title: Text(locale.fromFilter,
+                            style: AppTextStyle.darkGreySize16),
+                        onTap: () {
+                          searchController.text = "${locale.fromFilter}@";
+                          searchFocusNode.requestFocus();
+                          model.notifyListeners();
```

Comment:
There should be another way of doing this in the ViewModel instead of calling notifyListeners in the view?
Or is'nt there?

Reply:
No, there isn't — `searchController` and `searchFocusNode` are both only visible in the view 😪